src/microbit/test/test_display.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+import threading
+from unittest import mock
+
+from ..model import constants as CONSTANTS
+from ..model.display import Display
+from ..model.image import Image
+from .. import shim
+
+
+STR_A = "09900:90090:99990:90090:90090"
+STR_QUESTION_MARK = "09990:90009:00990:00000:00900"
+STR_EXCLAMATION_MARK = "09000:09000:09000:00000:09000:"
+STR_SIX = "00090:00900:09990:90009:09990"
+
+
+class TestDisplay(object):
+    def setup_method(self):
+        self.display = Display()
+
+    @pytest.mark.parametrize("x, y, brightness", [(1, 1, 4), (2, 3, 6), (4, 4, 9)])
+    def test_get_pixel(self, x, y, brightness):
+        self.display._Display__image._Image__LED[y][x] = brightness
+        assert brightness == self.display.get_pixel(x, y)
+
+    @pytest.mark.parametrize("x, y", [(5, 0), (0, -1), (0, 5)])
+    def test_get_pixel_error(self, x, y):
+        with pytest.raises(ValueError, match=CONSTANTS.INDEX_ERR):
+            self.display.get_pixel(x, y)
+
+    @pytest.mark.parametrize("x, y, brightness", [(1, 1, 4), (2, 3, 6), (4, 4, 9)])
+    def test_set_pixel(self, x, y, brightness):
+        self.display.set_pixel(x, y, brightness)
+        assert brightness == self.display._Display__image._Image__LED[y][x]
+
+    @pytest.mark.parametrize(
+        "x, y, brightness, err_msg",
+        [
+            (5, 0, 0, CONSTANTS.INDEX_ERR),
+            (0, -1, 0, CONSTANTS.INDEX_ERR),
+            (0, 0, -1, CONSTANTS.BRIGHTNESS_ERR),
+        ],
+    )
+    def test_set_pixel_error(self, x, y, brightness, err_msg):
+        with pytest.raises(ValueError, match=err_msg):
+            self.display.set_pixel(x, y, brightness)
+
+    def test_clear(self):
+        self.display._Display__image._Image__LED[2][3] = 7
+        self.display._Display__image._Image__LED[3][4] = 6
+        self.display._Display__image._Image__LED[4][4] = 9

```

Comment:
_&ast;nit&ast;_
I think I prefer `display.set_pixel` over `_Display__image._Image__LED`, since we don't expect users to directly access `__LED`. Up to you!

Reply:
agreed